### ISSUE TITLE
SD cant find installed repo, dosent see this extension

### ISSUE THREAD

**Issue by ronmurphy** [ModelSurge/sd-webui-comfyui#22]
I used Civitai helper to install your repo, and upon reload i get this error (i used python from powershell to run install_comfyui incase that was the issue)

>Error loading script: main.py
>Traceback (most recent call last):
>  File "D:\source\StabDiffCode\stable-diffusion-webui\modules\scripts.py", line 256, in load_scripts
    script_module = script_loading.load_module(scriptfile.path)
  File "D:\source\StabDiffCode\stable-diffusion-webui\modules\script_loading.py", line 11, in load_module
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\scripts\main.py", line 10, in <module>
    from lib_comfyui import webui_callbacks
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\webui_callbacks.py", line 4, in <module>
    from lib_comfyui import comfyui_adapter, webui_tab, webui_settings
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\webui_tab.py", line 70, in <module>
    <object data="http://127.0.0.1:{webui_settings.get_port()}" id="comfyui_webui_root"></object>
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\webui_settings.py", line 29, in get_port
    return settings_port or shared.cmd_opts.comfyui_port
AttributeError: 'Namespace' object has no attribute 'comfyui_port'

i am guessing that for whatever reason it (automatic1111) isn't seeing your extension?  Any ideas on how to fix this?

**Comment by John-WL** [comment]
The fact you installed it manually might be the issue, yes. I'm not sure why it's doing this exactly, but it's probably going to work if you install it through the webui extension's tab. 

Then, once this extension is downloaded (and not ComfyUI itself), you can go into the ComfyUI tab to install it from there. 

Let me know if doing this instead works or not.

**Comment by ronmurphy** [comment]
ok so it installed, but ni now get the following error...

>Traceback (most recent call last):
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ComfyUI\execution.py", line 182, in execute
    executed += recursive_execute(self.server, prompt, self.outputs, x, extra_data)
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ComfyUI\execution.py", line 58, in recursive_execute
    executed += recursive_execute(server, prompt, outputs, input_unique_id, extra_data)
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ComfyUI\execution.py", line 58, in recursive_execute
    executed += recursive_execute(server, prompt, outputs, input_unique_id, extra_data)
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ComfyUI\execution.py", line 67, in recursive_execute
    outputs[unique_id] = getattr(obj, obj.FUNCTION)(**input_data_all)
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ComfyUI\nodes.py", line 743, in sample
    return common_ksampler(model, seed, steps, cfg, sampler_name, scheduler, positive, negative, latent_image, denoise=denoise)
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ComfyUI\nodes.py", line 674, in common_ksampler
    model_management.load_model_gpu(model)
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ComfyUI\comfy\model_management.py", line 153, in load_model_gpu
    accelerate.dispatch_model(real_model, device_map=device_map, main_device="cuda")
  File "D:\source\StabDiffCode\stable-diffusion-webui\venv\lib\site-packages\accelerate\big_modeling.py", line 212, in dispatch_model
    check_device_map(model, device_map)
  File "D:\source\StabDiffCode\stable-diffusion-webui\venv\lib\site-packages\accelerate\utils\modeling.py", line 581, in check_device_map
    raise ValueError(
ValueError: The device_map provided does not give any device for the following parameters: betas, alphas_cumprod, alphas_cumprod_prev, sqrt_alphas_cumprod, sqrt_one_minus_alphas_cumprod, log_one_minus_alphas_cumprod, sqrt_recip_alphas_cumprod, sqrt_recipm1_alphas_cumprod, posterior_variance, posterior_log_variance_clipped, posterior_mean_coef1, posterior_mean_coef2

currently i have cuda installed, videocard is a 1650 gtx, and when i run comfyui alone, it works, maybe i have to pass along something like xformers, --no-half-vae, etc?

**Comment by ljleb** [comment]
This is probably not it, but can you verify that comfyui works if started by itself? (i.e. with the webui shutdown)
It looks like there's a problem with model loading but I am not sure what causes this.

**Comment by ronmurphy** [comment]
ok so from the comfyui lib folder that it is installed to, how do i start the app?  i tried python main.py and nope, no beans.

**Comment by ronmurphy** [comment]
ok, so i killed python, rebooted, and tried the webui again with comfy, first thing off the bat so i would know the right error messages, and this is what i got.

>Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is None and using 8 heads.
Setting up MemoryEfficientCrossAttention. Query dim is 320, context_dim is 768 and using 8 heads.
DiffusionWrapper has 859.52 M params.
Traceback (most recent call last):
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ComfyUI\execution.py", line 182, in execute
    executed += recursive_execute(self.server, prompt, self.outputs, x, extra_data)
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ComfyUI\execution.py", line 58, in recursive_execute
    executed += recursive_execute(server, prompt, outputs, input_unique_id, extra_data)
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ComfyUI\execution.py", line 58, in recursive_execute
    executed += recursive_execute(server, prompt, outputs, input_unique_id, extra_data)
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ComfyUI\execution.py", line 67, in recursive_execute
    outputs[unique_id] = getattr(obj, obj.FUNCTION)(**input_data_all)
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ComfyUI\nodes.py", line 743, in sample
    return common_ksampler(model, seed, steps, cfg, sampler_name, scheduler, positive, negative, latent_image, denoise=denoise)
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ComfyUI\nodes.py", line 674, in common_ksampler
    model_management.load_model_gpu(model)
  File "D:\source\StabDiffCode\stable-diffusion-webui\extensions\sd-webui-comfyui\lib_comfyui\ComfyUI\comfy\model_management.py", line 153, in load_model_gpu
    accelerate.dispatch_model(real_model, device_map=device_map, main_device="cuda")
  File "D:\source\StabDiffCode\stable-diffusion-webui\venv\lib\site-packages\accelerate\big_modeling.py", line 212, in dispatch_model
    check_device_map(model, device_map)
  File "D:\source\StabDiffCode\stable-diffusion-webui\venv\lib\site-packages\accelerate\utils\modeling.py", line 581, in check_device_map
    raise ValueError(
ValueError: The device_map provided does not give any device for the following parameters: betas, alphas_cumprod, alphas_cumprod_prev, sqrt_alphas_cumprod, sqrt_one_minus_alphas_cumprod, log_one_minus_alphas_cumprod, sqrt_recip_alphas_cumprod, sqrt_recipm1_alphas_cumprod, posterior_variance, posterior_log_variance_clipped, posterior_mean_coef1, posterior_mean_coef2

**Comment by John-WL** [comment]
The error you got comes from the webui's venv. Webui's venv is supposed to be a superset of ComfyUI's venv, but it looks like some dependancies are missing (?). 

One thing you can try to do is simply to delete the webui's venv, relaunch the webui to redownload it from scratch. 

It happened twice to me that my venv got messed up, and albeit I don't think deleting it to reinstall it is the recommended way to do this, it fixed stuff twice for me so far. 

This error is really confusing me to be honest 😅

**Comment by John-WL** [comment]
The error you got comes from the webui's venv. Webui's venv is supposed to be a superset of ComfyUI's venv, but it looks like some dependancies are missing (?). 

One thing you can try to do is simply to delete the webui's venv, relaunch the webui to redownload it from scratch. 

It happened twice to me that my venv got messed up, and albeit I don't think deleting it to reinstall it is the recommended way to do this, it fixed stuff twice for me so far. 

This error is really confusing me to be honest 😅

To be more precise, it looks like ComfyUI isn't generating the proper model object when it uses the webui's venv, and it totally doesn't do this on either my or ljleb's side. We can't reproduce.

**Comment by ronmurphy** [comment]
ok, so what do you need for me to do, should i try loading different safetensors?  i am using the hardlinked folders so that i don't have to duplicate my files, i was wondering if that was the case, i will remove the hardlinks and try it by way of the files in their folders

**Comment by ronmurphy** [comment]
do you think that this would have anything to do with it?

set COMMANDLINE_ARGS=--xformers --theme=dark --medvram --no-half-vae --api

? using openCanvas that needs the --api, and my card is a mobile 1650, hence everything but --theme=dark.

**Comment by John-WL** [comment]
I just tested with the parameters you are using, it doesn't seem to break anything. If you want to pass specific comfyui arguments, you can use the following for webui (they are being forwarded when launching comfyui from webui):
--comfyui-port
--comfyui-listen
--comfyui-dont-upcast-attention
--comfyui-use-split-cross-attention
--comfyui-use-pytorch-cross-attention
--comfyui-disable-xformers
--comfyui-highvram
--comfyui-normalvram
--comfyui-lowvram
--comfyui-novram
--comfyui-cpu

They are not yet in the docs of the repo, but they're implemented

**Comment by ronmurphy** [comment]
ok, sorry for vanishing, but i had the thought that i should provide a listing of the files in my \venv\ folder as how it's something missing that yall have installed but i dont, so, since it's over 6000 items .. sorry ...  use a diff tool maybe?  here is a link to my listing if yall want to check.

https://gist.github.com/ronmurphy/41c8a3edd00253adec3a70131048b638

**Comment by angrysky56** [comment]
error in the auto installed from automatic1111 extension, on restart after install- Did I do something wrong?
 D:\automatic1111\extensions\sd-webui-comfyui\ComfyUI

usage: main.py [-h] [--data-dir DATA_DIR] [--precision {full,autocast}] [--medvram] [--lowvram] [--lowram]
main.py: error: unrecognized arguments: --listen 127.0.0.1 --port 8188

**Comment by John-WL** [comment]
This error confuses me, --listen and --port are definitely part of the args in ComfyUI. Are you sure comfy is up to date?

**Comment by ljleb** [comment]
The original issue related to `shared.cmd_args` missing some properties seems to have been resolved. The follow-up issue related to extra model keys in the state dict should also be resolved now that we only pass the appropriate keys to comfyui. For these reasons, I'll close this issue.

@angrysky56 I'd appreciate if you could open another issue with a more thorough description of your setup! It would help contributors with debugging what you're running into.

**Comment by angrysky56** [comment]
I just did a fresh install- working now thanks- not sure what I had wrong, deleted automatics venv and reinstalled in automatic1111 extensions directly again. Instead of using my old nodes in my original install or copying them to the extension install I installed the manager first and reinstalled all the nodes that way, working fine. I had issues with the old comfyUI extension and there may have been some residue in automatics venv?
